gcp/appengine/blog/content/posts/introducing-broad-c-c++-support/index.md

```diff
@@ -0,0 +1,89 @@
+5. Determine the most recent commits for each submodule `git submodule status` 
+
+```
+git clone --recursive https://github.com/charlesneimog/pd-server
+cd pd-server
+git checkout cf3f15a841ca21b53c6de654c9981a30ae0b590c
+git submodule update
+git submodule status
+ 5c2e137f7a7a03f4007494954ccb3e23753e7807 pd-lib-builder (v0.6.0-28-g5c2e137)
+ 227d2c20509f85a394133e2be6d0b0fc1fda54b2 src/cpp-httplib (v0.11.3-6-g227d2c2)
+ 4c6cde72e533158e044252718c013a48bcff346c src/json (v3.11.2-39-g4c6cde72)
+ 1b11fd301531e6df35a6107c1e8665b1e77a2d8e src/websocketpp (0.8.2-1-g1b11fd3)
+```
+
+Cpp-httplib’s most recent commit hash is `227d2c20509f85a394133e2be6d0b0fc1fda54b2`. We can now use this information to construct an API call. It will be in this form:
+
+```
+curl -d \
+  '{"commit": "227d2c20509f85a394133e2be6d0b0fc1fda54b2"}' \
+  "https://api.osv.dev/v1/query" | jq '.vulns | map(.id)'
+```
+ Which returns:
+```
+[
+  "CVE-2023-26130"
+]
+```
+This result shows that the pd-server project is vulnerable to [CVE-2023-26130](https://osv.dev/vulnerability/CVE-2023-26130) at git commit ‘cf3f15a841ca21b53c6de654c9981a30ae0b590c’ through its use of cpp-httplib. Fortunately cpp-httplib has [a fix](https://github.com/yhirose/cpp-httplib/commit/5b397d455d25a391ba346863830c1949627b4d08) and pd-server updated their copy of cpp-httplib. The pd-server project is no longer be vulnerable to CVE-2023-26130. 
+
+### Vendored C/C++ dependencies
+
+Vendored dependencies are included in a project by simply copying the code into the repository. Git commit information is not retained, so we need another way to determine whether a vulnerability is present. 
+
+The determineversion API estimates your dependency’s version by comparing files hashes from your local project to known hashes for a given version. Once your dependency version is known, you can find the relevant vulnerabilities through searching our database or using our API.
+
+When we released the API in July, its use was limited to vulnerabilities found by [OSS-Fuzz](https://google.github.io/oss-fuzz/). Not all C/C++ projects are part of OSS-Fuzz, nor are all vulnerabilities for a given dependency found by OSS-Fuzz, so a number of vulnerabilities were left on the table. With the addition of the commit level vulnerability data from the NVD, this gap has been significantly narrowed. **This means that the determineversion API may now be used for the majority of vendored C/C++ dependencies.** 
+
+For more information on how to use the determineversion API, please see our [documentation](https://google.github.io/osv.dev/post-v1-determineversion/) or this [walkthrough](https://google.github.io/osv.dev/post-v1-determineversion/). 
+
+Within the next few months, support will be added to OSV-Scanner to make this a seamless out of the box experience for developers. Follow [this issue](https://github.com/google/osv-scanner/issues/82) for updates.
```

Comment:
We can drastically change this (and the example earlier) if we can merge https://github.com/google/osv-scanner/pull/581 prior to publication (currently dependent on merging https://github.com/go-git/go-git/pull/894)...

Reply:
I think it's likely we'll get this in prior to publication? (@another-rex we might just need another release?) 

We should also mention determineversions support coming soon as well, and we can drop a very short post when that lands too (https://github.com/google/osv-scanner/issues/612)

Reply:
Yes, it's merged, we just need a release of OSV Scanner that includes it.